server/src/main/java/org/elasticsearch/action/admin/cluster/node/shutdown/NodesRemovalPrevalidation.java

```diff
@@ -119,7 +121,7 @@ public static NodeResult fromXContent(XContentParser parser) throws IOException
     }
 
     // The prevalidation result of a node
-    public record Result(boolean isSafe, Reason reason, String message) implements ToXContentObject, Writeable {
+    public record Result(boolean isSafe, @Nullable Reason reason, String message) implements ToXContentObject, Writeable {
```

Comment:
Can we add a comment that `reason` is only nullable for bwc, i.e., for a fully upgraded cluster, we expect non-null?